cmd/rokka/cli/copyall/copyall.go

```diff
@@ -0,0 +1,128 @@
+package copyall
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/rokka-io/rokka-go/rokka"
+)
+
+//Options general options for the whole copy-all command
+type Options struct {
+	SourceOrganization      string
+	DestinationOrganization string
+	DryRun                  bool
+	Concurrency             int
+}
+
+type listReturn struct {
+	Cursor     string
+	ItemsCount int
+}
+
+//CopyResult the results of the copy operation
+type CopyResult struct {
+	RokkaHash string
+	Error     error
+}
+
+//StartWorkers starts Copy Workers
+func StartWorkers(options Options, client *rokka.Client, images chan string, results chan CopyResult, quit chan bool) {
+	waitGroup := sync.WaitGroup{}
+	waitGroup.Add(options.Concurrency)
+
+	// Start workers for image copy
+	for i := 0; i < options.Concurrency; i++ {
+		go func() {
+			defer waitGroup.Done()
+			copyWorker(client, images, results, quit, options.SourceOrganization, options.DestinationOrganization, options.DryRun)
+		}()
+	}
+
+	// Start a go-routine to to close the result channel as soon as all workers are done
+	go func() {
+		waitGroup.Wait()
+		close(results)
+	}()
+
+}
+
+func copyWorker(
+	client *rokka.Client,
+	imageFiles chan string,
+	results chan CopyResult,
+	quit chan bool,
+	sourceOrg string,
+	destinationOrg string,
+	dryRun bool,
+) {
+	for hash := range imageFiles {
+		result := CopyResult{
+			RokkaHash: hash,
+		}
+
+		if dryRun {
+			result.RokkaHash = "DRY-RUN-HASH"
+		} else {
+			err := executeRokkaCopy(client, hash, sourceOrg, destinationOrg)
+			if err != nil {
+				result.Error = err
+			}
+		}
+
+		select {
+		case results <- result:
+		case <-quit: // If we get a quit message, end the worker!
+			return
+		}
+	}
+}
+
+//Scan starts the scan operation for getting all images
+func Scan(options Options, client *rokka.Client, images chan string, quit chan bool) error {
+	defer func() {
+		close(images)
+	}()
+	doIt := true
+	cursor := ""
+	for doIt {
+		listReturnValues, err := list(options, client, images, quit, cursor)
+		if err != nil {
+			return err
+		}
+
+		if listReturnValues.Cursor == "" || cursor == listReturnValues.Cursor || listReturnValues.ItemsCount == 0 {
+			doIt = false
+		} else {
+			cursor = listReturnValues.Cursor
+		}
+	}
+	return nil
+}
+
+func list(options Options, client *rokka.Client, images chan string, quit chan bool, cursor string) (listReturn, error) {
+	listSourceImagesOptions := rokka.ListSourceImagesOptions{}
+	listReturnValues := listReturn{"", 0}
```

Comment:
I'm not fan of the `listReturn*` naming. The type is only used for communication between `scan` and `list` so I'd instead just return 3 values: `cursor` `imageCount` and `error`. 
You can even name the return values, if that makes code easier to write and read:
`(cursor string, imageCount string, err error)`

Reply:
changed it. Didn't get what you mean with name return values. I guess it's not that https://tour.golang.org/basics/7 ;)

Reply:
it is that ;)

Reply:
but it's ok like you did it now, too.